clients/lsp-lua.el

```diff
@@ -530,19 +530,19 @@ and `../lib` ,exclude `../lib/temp`.
    ("Lua.completion.callSnippet" lsp-lua-completion-call-snippet)
    ("Lua.color.mode" lsp-lua-color-mode)))
 
-(defun lsp-lua-language-server-install (client callback error-callback update?)
+(defun lsp-lua-language-server-install-latest (client callback error-callback update?)
   "Download the latest version of lua-language-server and extract it to
 `lsp-lua-language-server-install-dir'."
   (ignore client update?)
-  (let ((store-path (expand-file-name "vs-lua" lsp-clients-lua-language-server-install-dir)))
+  (let ((store-path (expand-file-name "lua-language-server-github" lsp-clients-lua-language-server-install-dir)))
     (lsp-download-install
      (lambda (&rest _)
        (set-file-modes lsp-clients-lua-language-server-bin #o0700)
        (funcall callback))
      error-callback
-     :url (lsp-vscode-extension-url "sumneko" "lua" "1.17.4")
+     :url (lsp--find-latest-release-url "https://api.github.com/repos/sumneko/lua-language-server/releases/latest")
```

Comment:
nit: also less generic but you can get the latest Github version via `https://github.com/sumneko/lua-language-server/releases/latest/download/lua-language-server-3.5.3-win32-x64.zip` for example.